providers/targeted_refresh.md

```diff
@@ -0,0 +1,180 @@
+
+There are two main differences between targeted refresh and full refresh:
+
+1. What is collected
+2. How inventory is deleted
+
+Targeted refresh only fetches inventory from the provider for the targets requested.  This cuts down on API calls and uses less memory.
+
+As for deletions, the most significant difference with targeted refresh is the "scope".  A full refresh considers the entire association (e.g.: `ems.vms.all`) to be in scope, so if something is in the association but not in the collected inventory it will be deleted.
+
+Targeted refresh requires scope to be limited, otherwise everything besides what you targeted would be deleted.  Most of this is abstracted away from the provider author but it is important to understand the fundamentals.  This is covered in way more detail in the [Provider Refresh](refresh.md) guide.
+
+### Getting Started
+
+The first thing to do is to set up the new classes that we need.  We'll need to add a `TargetCollection` class under the `Collector` and `Persister`.
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Collector::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Collector
+end
+```
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Parser::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Parser
+end
+```
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Persister::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Persister
+  # Indicate that this is a targeted refresh
+  def targeted?
+    true
+  end
+
+  # This tells the Persister what to do with references, specifically this will
+  # lookup references for inventory objects which aren't in the collection which
+  # is required for targeted refresh.
+  def strategy
+    :local_db_find_missing_references
+  end
+end
+```
+
+And don't forget to add `require_nested :TargetCollection` to `app/models/manageiq/providers/awesome_cloud/inventory/collector.rb`, `app/models/manageiq/providers/awesome_cloud/inventory/parser.rb`, and `app/models/manageiq/providers/awesome_cloud/inventory/persister.rb`
+
+We also have to add a configuration setting to enable/disable targeted refresh in your provider's `config/settings.yml`:
+```
+:ems_refresh:
+  :awesome_cloud:
+    :allow_targeted_refresh: true
+```
+
+We should now have all of the scaffolding in place to start implementing the targeted collector.  The goal is that the parser and persister are basically identical in full and targeted refresh and only the collector has to change.
+
+### Targeted Collection
+
+Now that we have the basics down we can begin writing the targeted collector.  Again this is where the majority of the work will be as long as we've done everything correctly.
+
+First we have to override each collection that is defined in the base Collector:
```

Comment:
This is an interesting step... if someone adds a new method to the base, they have to remember to "stub" it in the targeted?  That feels error prone, and backwards, somehow.  Not sure how to "fix" that, though.

Reply:
Yeah I agree, as I was writing this I started looking at moving the "full" out of the base collector class so that targeted isn't inheriting from it and accidentally call the full method.

Reply:
All too often, explaining something clarifies things not only for the audience, but for the explainer as well.  The more of these guides we write, the more pluggable our providers will become as we all notice code that seems to be needless or unintuitive.